File: Difficulty: Easy/Max Value Permutation/max-value-permutation.cpp
#define ll long long
#define mod ((int)(1e9+7))

class Solution {
  public:
    int maxValue(vector<int> &arr) {
        // Complete the function
        sort(arr.begin(),arr.end());
        ll ans = 0;
        int n = arr.size();
        for(int i = n-1;i>=0;i--) {
            ans = (ans + 1LL * i * arr[i])%mod;
        }
        return ans;
    }
};

File: Difficulty: Easy/Max Value Permutation/max-value-permutation.java
// User function Template for Java

class Solution {

    int maxValue(int arr[]) {
        // Complete the function
        int mod = 1000000007;
        Arrays.sort(arr);
        int n = arr.length;
        int ans = 0;
        for(int i = 1;i<n;i++) {
            ans = (int) ((ans + (long) i * arr[i]) % mod);
        }
        
        return ans;
    }
}
